### ISSUE TITLE
PostgreSQL exporter delete-task does not seem to be working 

### ISSUE THREAD

**Issue by PSjoe** [algorand/conduit#117]
### The PostgreSQL exporter `delete-task` option does not seem to be working

We were hoping to use the `delete-task` option to keep our dev databases a bit smaller.  Our `conduit.yml` file looks like this:  
```
exporter:
  name: postgresql
  config:
    connection-string: <redacted>
      sslmode=disable
    delete-task:
      rounds: 1000
      interval: 20
```

However, when running against either a fully-synced PostgreSQL instance or an instance that is starting from 0, it does not appear that the `txn` table is being culled.  Using the above configs, I can see the following in the DB: 

```
betanet=> select round from txn;
 round 
-------
   357
 12166
 12178
 12188
 13011
 13018
 13020
 14810
 14817
 14823
 14864
 15655
 15661
 15668
 18075
 36998
(16 rows)
```

I'd have expected the early rows to have been thrown out as conduit moves through later blocks.

### Your environment

* Conduit 1.2.0 (37a13fe)
* PostgreSQL 13.10
* Algod 3.16.3.stable [rel/stable] (commit #cbccc6de)
* Ledger:  Betanet

### Steps to reproduce

1. Bootstrap an empty PosgreSQL DB 
2. Start Conduit using the PostgreSQL exporter from block 0 using the configs above.

### Expected behaviour
Only transactions within the last 1000 blocks would be present 

### Actual behaviour
All transactions appear to be present.  Table continues to get longer as conduit works its way through the ledger.

**Comment by tzaffi** [comment]
Another useful data point for our investigation is how long the experiment ran for. If it was only a couple of seconds long, it may not have triggered a pruning of the database since that's on a timer that runs every 2 seconds.

**Comment by PSjoe** [comment]
Config file is as follows: 
```
log-level: INFO

# If no log file is provided logs are written to stdout.
#log-file:

# Set to 0 to retry forever.
retry-count: 10

retry-delay: "1s"


# Whether or not to print the conduit banner on startup.
hide-banner: false

# When enabled prometheus metrics are available on '/metrics'
metrics:
    mode: ON
    addr: ":9999"
    prefix: "conduit"

# The importer is typically an algod follower node.
importer:
    name: algod
    config:
        mode: "follower"
        netaddr: http://localhost:48081
        token: <redacted>
        catchup-config:
            auto: true
            admin-token: <redacted>


# Zero or more processors may be defined to manipulate what data
# reaches the exporter.
processors:

# An exporter is defined to do something with the data.
exporter:
  name: postgresql
  config:
    connection-string: <redacted>
    delete-task:
      rounds: 1000
      interval: 20
```

There were no "DeleteLoop" logs that I was able to see.  I didn't save the logs and unfortunately didn't keep the setup around.

I let it run over lunch (~1h).

**Comment by tzaffi** [comment]
Thanks for the info! In, particular:

> log-level: INFO

explains why you don't see "DeleteLoop" logs. I recommend trying again with log-level `WARN`.

**Comment by tzaffi** [comment]
Apologies. That last suggestion is incorrect as we should still have seen warnings under the `INFO` log level. I'll do some further investigation.

**Comment by PSjoe** [comment]
Running with log-level `WARN` is giving me even less logging then with INFO.

**Comment by tzaffi** [comment]
> There were no "DeleteLoop" logs that I was able to see. I didn't save the logs and unfortunately didn't keep the setup around.

Since you didn't save the logs, I'm back to my original suggestion, re-running the experiment and reporting all `WARN`'s which should hopefully have a "DeleteLoop" log.

**Comment by PSjoe** [comment]
There _no_ log messages at all at log level WARN: 
```
kubectl logs vira-betanet-blue-conduit-conduit-0 --follow 
Defaulted container "conduit" out of: conduit, create-readonly-user (init), copy-config-files (init)
Writing logs to console.

```
It's been running now for ~5 minutes.

**Comment by PSjoe** [comment]
I'm afraid I don't have the runway to re-create the testbed any more.  I've got a dev environment with the following configs:  
```
exporter:
  config:
    connection-string: <redacted>
  
    delete-task:
      interval: 1000
      rounds: 10000
  name: postgresql

hide-banner: true
importer:
  config:
    catchup-config:
      admin-token: <redacted>
    mode: follower
    netaddr: http://algod4conduit:48081
    token: <redacted>
  name: algod
log-level: "INFO"
metrics:
  addr: ":9999"
  mode: "ON"
  prefix: "conduit"
retry-count: 0
retry-delay: 1s
```

It has been running for the last 24-ish hours and no log messages with `DeleteLoop` are in our log collection system.  This running against a BetaNet database that was brought up using a snap-shot and Conduit then brought the PostgreSQL DB fully up to date.

**Comment by PSjoe** [comment]
If there are any errors with the config file, they are not being mentioned when conduit starts.  The first set of INFO log messages look like this:  
```
{"__type":"Conduit","_name":"main","level":"info","msg":"Starting Conduit Conduit 1.2.0 (37a13fe)","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Using data directory: /conduit","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Conduit configuration is valid","time":"2023-07-18T17:47:02Z"}
Writing logs to console.
{"__type":"Conduit","_name":"main","level":"info","msg":"Found Importer: algod","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Found Exporter: postgresql","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Starting Pipeline Initialization","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Checking for round override","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Overriding default next round from 0 to 28056796.","time":"2023-07-18T17:47:02Z"}
{"__type":"importer","_name":"algod","level":"info","msg":"No catchup required to reach round 28056796","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Initialized Importer: algod","time":"2023-07-18T17:47:02Z"}
{"__type":"exporter","_name":"postgresql","level":"info","msg":"Running 0 migrations.","time":"2023-07-18T17:47:02Z"}
{"__type":"exporter","_name":"postgresql","level":"info","msg":"Setting status: Migrations Complete","time":"2023-07-18T17:47:02Z"}
{"__type":"exporter","_name":"postgresql","level":"info","msg":"Migration finished successfully.","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Initialized Exporter: postgresql","time":"2023-07-18T17:47:02Z"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Pipeline round: 28056796","time":"2023-07-18T17:47:02Z"}
{"__type":"exporter","_name":"postgresql","level":"info","msg":"adding block 28056796","time":"2023-07-18T17:47:02Z"}
```

**Comment by tzaffi** [comment]
Update: I can reproduce the issue locally and am working on a fix. Stay tuned.

**Comment by tzaffi** [comment]
Fix coming in #119

**Comment by PSjoe** [comment]
confirmed working!  Thanks!
